unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -62,36 +62,39 @@ public class PhysicsRemoteFPSAgentController : BaseFPSAgentController {
             return ItemInHand;
         }
 
-        // get all sim objets of action.type, then sets their temperature decay timers to value
-        public void SetRoomTempDecayTimeForType(string objectType, float TimeUntilRoomTemp = 0.0f) {
-            // get all objects of type passed by action
-            SimObjPhysics[] simObjects = GameObject.FindObjectsOfType<SimObjPhysics>();
-
-            List<SimObjPhysics> simObjectsOfType = new List<SimObjPhysics>();
-
-            foreach (SimObjPhysics sop in simObjects) {
-                if (sop.Type.ToString() == objectType) {
-                    simObjectsOfType.Add(sop);
-                }
+        public void EnableTemperatureDecay() {
+            if (!physicsSceneManager.GetComponent<PhysicsSceneManager>().AllowDecayTemperature) {
+                physicsSceneManager.GetComponent<PhysicsSceneManager>().AllowDecayTemperature = true;
             }
-            // use SetHowManySecondsUntilRoomTemp to set them all
-            foreach (SimObjPhysics sop in simObjectsOfType) {
-                sop.SetHowManySecondsUntilRoomTemp(TimeUntilRoomTemp);
+            actionFinished(true);
+        }
+
+        public void DisableTemperatureDecay() {
+            if (physicsSceneManager.GetComponent<PhysicsSceneManager>().AllowDecayTemperature) {
+                physicsSceneManager.GetComponent<PhysicsSceneManager>().AllowDecayTemperature = false;
             }
+            actionFinished(true);
+        }
 
+        // sets temperature decay for a single object.
+        public void SetTemperatureDecayTime(string objectId, float decayTime) {
+            if (decayTime < 0) {
+                throw new ArgumentOutOfRangeException("decayTime must be >= 0. You gave " + decayTime);
+            }
+            SimObjPhysics sop = getTargetObject(objectId: objectId, forceAction: true);
+            sop.SetHowManySecondsUntilRoomTemp(decayTime);
             actionFinished(true);
         }
 
-        // get all sim objects and globally set the room temp decay time for all of them
-        public void SetGlobalRoomTempDecayTime(float TimeUntilRoomTemp = 0.0f) {
-            // get all objects 
+        // globally sets temperature decay for all objects.
+        public void SetTemperatureDecayTime(float decayTime) {
+            if (decayTime < 0) {
+                throw new ArgumentOutOfRangeException("decayTime must be >= 0. You gave " + decayTime);
+            }
             SimObjPhysics[] simObjects = GameObject.FindObjectsOfType<SimObjPhysics>();
-
-            // use SetHowManySecondsUntilRoomTemp to set them all
             foreach (SimObjPhysics sop in simObjects) {
-                sop.SetHowManySecondsUntilRoomTemp(TimeUntilRoomTemp);
+                sop.SetHowManySecondsUntilRoomTemp(decayTime);
             }
-
             actionFinished(true);
         }
```

Comment:
This is basically everything, in terms of new actions being added.

The rest of it is a find and replace with `canChangeTempToHot` to `isHeatSource` and `canChangeTempToCold` to `isColdSource`.